### PR TITLE
feat/admin panel

### PR DESCRIPTION
**Pull request by czaja307**

### CODE REVIEW

**Comment by czaja307** [comment]
image

i can't create user with create:user command, the adonis doesn't seem to recognize the hashing fuctions, even though i used the default setup shown in documentation

@dawidlinek help much appreciated

File: commands/create_user.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  @args.string({
+    argumentName: 'password',
+    description: 'Password for the user',
+  })
+  declare password: string
+
+  async run() {
+    this.logger.info(`Creating user: ${this.email}`)
+    await User.create({
```
Comment (czaja307) [inline commands/create_user.ts:22]: i'm trying to create user here

File: app/models/user.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { DateTime } from 'luxon'
+import hash from '@adonisjs/core/services/hash'
+import { compose } from '@adonisjs/core/helpers'
+import { BaseModel, column } from '@adonisjs/lucid/orm'
+import { withAuthFinder } from '@adonisjs/auth/mixins/lucid'
+
+const AuthFinder = withAuthFinder(() => hash.use('scrypt'), {
```
Comment (czaja307) [inline app/models/user.ts:7]: adonis doesn't seem to be recognising this

File: commands/create_user.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { args, BaseCommand } from '@adonisjs/core/ace'
+import User from '#models/user'
+
+export default class CreateUser extends BaseCommand {
+  static commandName = 'create:user'
+  static description = 'Create a new user in the system'
+
```
Comment (dawidlinek) [inline commands/create_user.ts:7]: @czaja307 
Try adding startApp param here. It boots the application and enables more functionalities. 
```suggestion
static options: CommandOptions = {
    startApp: true
  }
```

More info here https://docs.adonisjs.com/guides/ace/creating-commands#command-metadata

File: app/admin/auth.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    return null
+  }
+}
+
+const authProvider = new DefaultAuthProvider({
+  componentLoader,
+  authenticate,
+})
+
+export default authProvider
```
Comment (simon-the-shark) [inline app/admin/auth.ts:32]: default exporty not nice xd

File: app/admin/resources/timestamps.ts
Comment (simon-the-shark) [inline app/admin/resources/timestamps.ts:1]: maybe somehow extract those files that are not standalone adminjs models to some utils folder or smth? idk

File: config/adminjs.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+const adminjsConfig: AdminJSProviderConfig = {
+  adapter: {
+    enabled: true,
+  },
+  adminjs: {
+    rootPath: '/admin',
+    loginPath: '/admin/login',
+    logoutPath: '/admin/logout',
+    componentLoader,
+    resources: [academicCalendarResource, buildingResource, campusResource],
```
Comment (simon-the-shark) [inline config/adminjs.ts:16]: if you want to have all "resources" in one folder, then you can drop this list to `index.ts` there?

**Review by simon-the-shark — COMMENTED** [review]
had a quick look, but didn't notice anything suspicious. The notes here are very **opinionated**

File: app/admin/auth.ts (hunk 4 above)
Comment (czaja307) [reply app/admin/auth.ts:32]: that is actually not my code xd, it was generated during setup

File: app/admin/resources/timestamps.ts
Comment (czaja307) [reply app/admin/resources/timestamps.ts:1]: gonna think about it, but this one is pretty much related to models imo

**Review by simon-the-shark — APPROVED** [review]
lgtm (except default exports :P)

**Pull request merged**